.github/workflows/secrets-scanner.yaml

```diff
@@ -51,18 +54,92 @@ jobs:
           head: ${{ github.ref_name }}
           extra_args: --debug --only-verified
 
+      - name: Notify Slack on Failure
+        if: ${{ failure() && (github.ref_name == 'master' || github.ref_name == 'release') }}
+        uses: act10ns/slack@ed1309ab9862e57e9e583e51c7889486b9a00b0f # v2.0.0
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+        with:
+          status: ${{ job.status }}
+          steps: ${{ toJson(steps) }}
+          channel: '#eng-dataset-o11y'
+
+  trufflehog_cron_master_branch:
+    name: TruffleHog - cron (master branch)
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'schedule' }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: "master"
+          fetch-depth: 0
+
+      # Special check which ensures that the clone performed above is not shallow. We need the
+      # complete git history for scanning to work correctly in all the situations. In some cases
+      # if a shallow clone is used, trufflehog won't not fail with an error, but it would simply
+      # not detect any files and that could be dangerous.
+      - name: Shallow repo check
+        run: |
+          if git rev-parse --is-shallow-repository | grep -q "true"; then
+            echo "Encountered a shallow repository, trufflehog may not work as expected!"
+            exit 1
+          fi
+
+      # As part of cron trigger we scan the whole repo directory.
+      # NOTE: Since trufflehog GHA is meant to be used in context of push / pr it can't be
+      # used dorectly to scan the whole repo directory. This may take a while, but it's good idea
+      # to run it on a daily basis.
+      - name: scan-cron-master
+        run: |
+          docker run --rm -v "$PWD:/workdir" trufflesecurity/trufflehog:latest git \
+            file:///workdir --fail --no-update --debug --only-verified
+
+      - name: Notify Slack on Failure
+        if: ${{ failure() && (github.ref_name == 'master' || github.ref_name == 'release') }}
+        uses: act10ns/slack@ed1309ab9862e57e9e583e51c7889486b9a00b0f # v2.0.0
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+        with:
+          status: ${{ job.status }}
+          steps: ${{ toJson(steps) }}
+          channel: '#eng-dataset-o11y'
+
+  trufflehog_cron_release_branch:
```

Comment:
trufflehog_cron_release_branch are identical, except for the ref parameter.
What about using strategy.matrix here?

Reply:
Thanks for the review.

That could work, but I wanted to use this workflow as-is (copy & paste) since we use it in many other places and right now we don't have public open source org level Github repository with re-usable actions and workflows where we could store and reference from workflows like that.

Reply:
But maybe that's also an opportunity to start such repo with public and re-usable actions.

I would be happy to do that, but we also need to follow the official process and get all the necessary approvals for that.